### PR TITLE
Add sample plugin for File System Provider API

### PR DESCRIPTION
**Pull request by azatsarynnyy**
The plugin demonstrates using a contributed `FileSystemProvider` for reading/writing a content with `workspace.openTextDocument` API.

demo-fs

The related PR https://github.com/theia-ide/theia/pull/4484

### CODE REVIEW

File: samples/fs-plugin/src/demo-fs-provider.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import * as theia from '@theia/plugin';
```
Comment (benoitf) [inline samples/fs-plugin/src/demo-fs-provider.ts]: missing copyright

File: samples/fs-plugin/src/fs-plugin.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (benoitf) [inline samples/fs-plugin/src/fs-plugin.ts]: missing copyright

**Comment by azatsarynnyy** [comment]
thanks @benoitf

**Comment by benoitf** [comment]
sorry, I forgot to apply review changes last time :-/

**Comment by azatsarynnyy** [comment]
np )

**Pull request merged**